### ISSUE TITLE
Rule Change: (no-throw-literal)

### ISSUE THREAD

**Issue by kosiakMD** [eslint/eslint#18055]
### What rule do you want to change?

no-throw-literal

### What change do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new option

### Example code

```js
'@typescript-eslint/no-throw-literal': ['error', {
      allowClass: ['ValidationError'], // or ignore/ignorePattern, e.g. '*Error'
    }]
```
```


### What does the rule currently do for this code?

Turn off the check for specific instances by class names.

### What will the rule do after it's changed?

It will ignore custom errors (which could not be extended from the Error).

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

In NestJS, there are Errors that are not extended from Error but are totally valid for the flow
```
    const error = new ValidationError();
    error.target = obj;
    error.property = field || key;
    error.value = fieldValue;
    error.constraints = {...};
    error.children = [];
    throw error;
```

**Comment by aladdin-add** [comment]
`@typescript-eslint/no-throw-literal` is not maintained by eslint team. please file the the issue in https://github.com/typescript-eslint/typescript-eslint.
